examples/parse-text-from-any-block-type/index.js

```diff
@@ -0,0 +1,148 @@
+    return caption + ": " + source
+  }
+  // If no caption, just return the source URL
+  return source
+}
+
+// Parse the plain text from any block type supported by the public API.
+const parseText = block => {
+  let text
+
+  // The public API does not support all block types yet
+  if (block.type === "unsupported") {
+    text = "Unsupported block type"
+  }
+  // Get rich text from blocks that support it
+  else if (block[block.type].rich_text) {
+    // This will be an empty string if it's an empty line.
+    text = getPlainTextFromRichText(block[block.type].rich_text)
+  }
+  // Get text for block types that don't have rich text
+  else {
+    switch (block.type) {
+      case "bookmark":
+        text = block.bookmark.url
+        break
+      case "child_database":
+        text = block.child_database.title
+        // Use "Query a database" endpoint to get db rows: https://developers.notion.com/reference/post-database-query
+        // Use "Retrieve a database" endpoint to get additional properties: https://developers.notion.com/reference/retrieve-a-database
+        break
+      case "child_page":
+        text = block.child_page.title
+        break
+      case "embed":
+      case "video":
+      case "file":
+      case "image":
+      case "pdf":
+        text = getMediaSourceText(block)
+        break
+      case "equation":
+        text = block.equation.expression
+        break
+      case "link_preview":
+        text = block.link_preview.url
+        break
+      case "synced_block":
+        // Provides ID for block it's synced with.
+        text = block.synced_block.synced_from
+          ? "This block is synced with a block with the following ID: " +
+            block.synced_block.synced_from[block.synced_block.synced_from.type]
+          : "Source sync block that another blocked is synced with."
+        break
+      case "table":
+        // Only contains table properties.
+        // Fetch children blocks for more details.
+        text = "Table width: " + block.table.table_width
+        break
+      case "table_of_contents":
+        // Does not include text from ToC; just the color
+        text = "ToC color: " + block.table_of_contents.color
+        break
+      case "breadcrumb":
+      case "column_list":
+      case "divider":
+        text = "No text available"
+        break
+      default:
+        text = "Needs case added"
+        break
+    }
+  }
+  // Blocks with the has_children property will require fetching the child blocks. (Not included in this example.)
+  // e.g. nested bulleted lists
+  if (block.has_children) {
+    // For now, we'll just flag there are children blocks.
+    text = text + " (Has children)"
+  }
+  // Includes block type for readability. Update formatting as needed.
+  return block.type + ": " + text
+}
+
+async function retrieveBlockChildren(id) {
+  console.log("Retrieving blocks (async)...")
+
+  const blocks = await notion.blocks.children.list({
```

Comment:
Should we include pagination here? We have a helper called `iteratePaginatedAPI` included in the SDK 🤷

Reply:
oh cool, thanks! I'll add it

Reply:
Added! Thanks for the review. 🙏 Looks like it needs another look to merge so I re-requested another one.